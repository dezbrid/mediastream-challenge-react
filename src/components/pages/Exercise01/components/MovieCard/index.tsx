import React from "react";
import "./assets/styles.css";
import { useAppDispatch } from "../../../../../app/hooks";

import { MovieCardInterfaces } from "../../interface";
import { addCart,incrementItemByID,decrementItemByID } from "../../cartSlice";

export default function MovieCard(props: MovieCardInterfaces) {
  const dispatch = useAppDispatch();

  return (
    <li className={"movies__card"}>
      <ul>
        <li>ID:{props.id}</li>
        <li>Name: {props.name}</li>
        <li>Price: ${props.price}</li>
      </ul>
      {props.quantity ? (
        <div className="movies__card-quantity">
          <button onClick={() => dispatch(decrementItemByID(props.id))}>-</button>
          <span>{props.quantity}</span>
          <button onClick={() => dispatch(incrementItemByID(props.id))}>+</button>
        </div>
      ) : (
        <button
          onClick={() => dispatch(addCart(props))}
          className="movies__card-add-cart"
          disabled={props.disableAddtoCart}
        >
          Add to cart
        </button>
      )}
    </li>
  );
}
