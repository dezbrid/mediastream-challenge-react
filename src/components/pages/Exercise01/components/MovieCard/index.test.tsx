import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MOVIES } from "../../mocks";
import MovieCard from "./index";
import { store } from '../../../../../app/store';

describe("<MovieCard />", () => {
  it("<MovieCard /> snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MovieCard {...MOVIES[0]} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
