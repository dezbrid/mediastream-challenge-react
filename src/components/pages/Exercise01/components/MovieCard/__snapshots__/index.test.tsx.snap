// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> <MovieCard /> snapshot 1`] = `
<li
  className="movies__card"
>
  <ul>
    <li>
      ID:
      1
    </li>
    <li>
      Name: 
      Star Wars
    </li>
    <li>
      Price: $
      20
    </li>
  </ul>
  <button
    className="movies__card-add-cart"
    onClick={[Function]}
  >
    Add to cart
  </button>
</li>
`;
